libsolidity/analysis/ConstantEvaluator.cpp

```diff
@@ -37,15 +37,15 @@ using namespace solidity::langutil;
 
 void ConstantEvaluator::endVisit(UnaryOperation const& _operation)
 {
-	auto sub = type(_operation.subExpression());
+	auto sub = evaluatedValue(_operation.subExpression());
```

Comment:
Maybe just `value`?

Reply:
Or just `evaluate` (whether or not it actually computes something or just returns the cached value is irrelevant, I would sy).